### PR TITLE
BOAC-2542, note_template API endpoints: my_templates, create

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2542

### CODE REVIEW

**Comment by paulkerschen** [comment]
A collaborative solution indeed.

**Comment by johncrossman** [comment]
@paulkerschen Sorry, did you mean to merge?

File: boac/api/note_templates_controller.py [hunk 1]
```diff
@@ -39,3 +67,22 @@ def get_note_template(note_template_id):
...
+@app.route('/api/note_template/delete/<note_template_id>', methods=['DELETE'])
+@login_required
+def delete_note_template(note_template_id):
+    note_template = NoteTemplate.find_by_id(note_template_id=note_template_id)
+    if not note_template:
+        raise ResourceNotFoundError('Template not found')
+    if note_template.creator_id != current_user.get_id():
+        raise ForbiddenRequestError(f'Template not available')
+    NoteTemplate.delete(note_template_id=note_template_id)
+    return tolerant_jsonify({'message': f'Note template {note_template_id} deleted'}), 200
```
Comment (johncrossman) [inline boac/api/note_templates_controller.py]: @paulkerschen Fyi, I amended this PR to add `/api/note_template/delete` support

**Comment by paulkerschen** [comment]
Thanks, @johncrossman. Long day.

**Pull request merged**